### PR TITLE
[Dataset] SBMMixtureDataset

### PR DESCRIPTION
**Pull request by HuXiangkun**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/data/sbm.py [hunk 1]
```diff
@@ -118,3 +143,22 @@ def collate_fn(self, x):
         deglg_batch = np.concatenate(deg_lg, axis=0)
         pm_pd_batch = np.concatenate([x + i * self._n_nodes for i, x in enumerate(pm_pd)], axis=0)
         return g_batch, lg_batch, degg_batch, deglg_batch, pm_pd_batch
+
+
+class SBMMixture(SBMMixtureDataset):
```
[the remark this answers is not in the document]
Comment (HuXiangkun) [reply python/dgl/data/sbm.py:148]: How do I add deprecation warning?

**Comment by hetong007** [comment]
A patch of this PR could resolve the CI issue: https://github.com/dmlc/dgl/pull/1782

File: examples/pytorch/cluster_gcn/cluster_gcn.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def main(args):
         val_mask = val_mask.cuda()
         test_mask = test_mask.cuda()
 
-    print(torch.cuda.get_device_name(0))
```
Comment (classicsong) [inline examples/pytorch/cluster_gcn/cluster_gcn.py:96]: How about directly delete the code

File: python/dgl/data/ppi.py [hunk 3]
```diff
@@ -1,127 +1,135 @@
...
-            self.graph = DGLGraph(nx.DiGraph(json_graph.node_link_graph(g_data)))
-            self.graph_id = np.load('{}/ppi/train_graph_id.npy'.format(self._dir))
+        _url = _get_dgl_url('dataset/ppi.zip')
+        super(PPIDataset, self).__init__(name='ppi',
+                                         url=_url,
+                                         raw_dir=raw_dir,
+                                         force_reload=force_reload,
+                                         verbose=verbose)
+
+    def process(self, root_path):
```
Comment (classicsong) [inline python/dgl/data/ppi.py:63]: Tong has removed the root_path input argument, we need to change our code according to his update.
```suggestion
    def process(self):
```

File: python/dgl/data/ppi.py [hunk 4]
```diff
@@ -1,127 +1,135 @@
...
+        feat_file = os.path.join(root_path, '{}_feats.npy'.format(self.mode))
+        graph_id_file = os.path.join(root_path, '{}_graph_id.npy'.format(self.mode))
+
+        g_data = json.load(open(graph_file))
+        self._labels = np.load(label_file)
+        self._feats = np.load(feat_file)
+        self.graph = DGLGraph(nx.DiGraph(json_graph.node_link_graph(g_data)))
+        graph_id = np.load(graph_id_file)
+
+        lo, hi = 1, 21
```
Comment (classicsong) [inline python/dgl/data/ppi.py:75]: Add some docstring for this 1, 21.

File: python/dgl/data/ppi.py [hunk 5]
```diff
@@ -1,127 +1,135 @@
...
+    def has_cache(self):
+        graph_path = os.path.join(self.save_path, '{}_dgl_graph.bin'.format(self.mode))
+        info_path = os.path.join(self.save_path, '{}_info.pkl'.format(self.mode))
+        return os.path.exists(graph_path) and os.path.exists(info_path)
+
+    def save(self):
+        graph_path = os.path.join(self.save_path, '{}_dgl_graph.bin'.format(self.mode))
+        info_path = os.path.join(self.save_path, '{}_info.pkl'.format(self.mode))
+        save_graphs(graph_path, self._graphs)
+        save_info(info_path, {'labels': self._labels, 'feats': self._feats, 'graph': self.graph})
```
Comment (classicsong) [inline python/dgl/data/ppi.py:100]: why we save self.graph in info?

File: python/dgl/data/sbm.py [hunk 6]
```diff
@@ -49,9 +53,12 @@ def sbm(n_blocks, block_size, p, q, rng=None):
     adj = sp.sparse.triu(a) + sp.sparse.triu(a, 1).transpose()
     return adj
 
-class SBMMixture:
-    """ Symmetric Stochastic Block Model Mixture
-    Please refer to Appendix C of "Supervised Community Detection with Hierarchical Graph Neural Networks" (https://arxiv.org/abs/1705.08415) for details.
+
+class SBMMixtureDataset(object):
```
Comment (classicsong) [inline python/dgl/data/sbm.py:57]: Is this a self-contained dataset that do not need download data?
You should derived from DGLDataset at least.

**Pull request merged**